scripts/solution-manager.sh

```diff
@@ -175,29 +182,52 @@ containsElement () {
   return 1
 }
 
-_add_solution() {
-    # Skip adding solutions if none is passed
-    [ ${#SOLUTIONS[@]} -lt 1 ] && return 0
+remove_solutions() {
+  delete=("$@")
+  for target in "${delete[@]}"; do
+    for i in "${!EXISTENT_SOLUTIONS[@]}"; do
+      if [[ "${EXISTENT_SOLUTIONS[i]}" = "$target" ]]; then
+        unset 'EXISTENT_SOLUTIONS[i]'
+      fi
+    done
+  done
+  # Rebuild the gaps in the array
+  for i in "${!EXISTENT_SOLUTIONS[@]}"; do
+    new_array+=( "${EXISTENT_SOLUTIONS[i]}" )
+  done
+  EXISTENT_SOLUTIONS=("${new_array[@]}")
+  unset new_array
+}
+
+_add_del_solution() {
+    # Skip adding/deleting solutions if none is passed
+    [ ${#SOLUTIONS_REMOVE[@]} -lt 1  ] && [ ${#SOLUTIONS_ADD[@]} -lt 1 ] && return 0
     # The use salt file.serialize merge require having full list
     # salt-call output example:
     # local: ["/tmp/solution1.iso", "/tmp/solution2.iso"]
     # parsed products:
     # ("/tmp/solution1.iso" "/tmp/solution2.iso")
-    local existent_solutions=()
     IFS=" " read -r -a \
-        existent_solutions <<< "$(salt-call --out txt slsutil.renderer \
+        EXISTENT_SOLUTIONS <<< "$(salt-call --out txt slsutil.renderer \
           string="{{ pillar.metalk8s.solutions.configured | join(' ') }}" | cut -d' ' -f2- | tr -d '{}' )"
-    for solution in "${SOLUTIONS[@]}"; do
-        if ! containsElement "'$solution'" \
-             "${existent_solutions[@]+"${existent_solutions[@]}"}"; then
-            existent_solutions+=("'$solution'")
-        fi
-    done
+    # Add new solutions
+    if [ ${#SOLUTIONS_ADD[@]} -ge 1 ]; then
+        for solution in "${SOLUTIONS_ADD[@]}"; do
+            if ! containsElement "'$solution'" \
+                 "${EXISTENT_SOLUTIONS[@]+"${EXISTENT_SOLUTIONS[@]}"}"; then
+                EXISTENT_SOLUTIONS+=("'$solution'")
+            fi
+        done
```

Comment:
To be consistent can you create the function `add_solutions`

Reply:
does it make sens to have a function with just `EXISTENT_SOLUTIONS+=("'$solution'")` ?

Reply:
The function will also do the `for` loop + the `if ! containsElement`.
So you will have something like
```
if "SOLUTIONS_ADD not empty" then
  add_soltuions SOLUTIONS_ADD
if "SOLTUIONS_REMOVE not empty" then
  remove_soltuions SOLUTIONS_REMOVE
```
Which is cleaner IMO